### DISCUSSION TITLE
Semester 4 Week 5

### DISCUSSION THREAD

**Discussion by Stev3eM** [SE-TINF22B6/CookHub#200] [Show and tell]
Hello backpackers,

This is our weekly discussion-blog to keep you up to date within our project progression.
Another week, another update for you:

### Frontend

✅ We redesigned the FoodForge page to make it look better and increase the user friendliness

![image](https://github.com/SE-TINF22B6/CookHub/assets/113697513/40184e4b-6f38-4351-9b53-59e8ae2d02a0)

✅ We implemented a component which opens a dialog popup with Carlos providing additional information/instruction to the current webpage or its functionality.

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/4c0d9963-706d-4363-9160-411655f0ed37" width="300" />
 <br>
<br>

✅ Created Settings Page which will allow the user to make changes of his username, password and profile picture and to delete his account in the future.

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1b9896ee-5e59-4b4b-80cc-fe04b969e0ec" width="400" />

<br>
<br>

✅ Reworked Profile Page which is planed to show specific user information, recipe favourites, own recipes and recipe history.

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/48d43e56-1473-489c-8910-45c907c16262" width="400" />

<br>
<br>

✅ Small design changes on NavBar to increase contrast and visibilty

![image](https://github.com/SE-TINF22B6/CookHub/assets/113697513/e19dbb16-43fb-4e09-abcb-7907d5f5966a)

✅ BugFixed Login Page by removing Material UI components and restructured with vanilla HTML and CSS.

<br>
<br>

## Testing

✅ We setup Selenium WebDriver for automated browser testing. Also see our [testing page](https://github.com/SE-TINF22B6/CookHub/wiki/Testing) in our wiki.

<br>
<br>

## Backend

✅ Implemented backend logic for login and authentication functionality including:

- LogIn
- LogOut
- IsLoggedIn
- Register
- Regexes used to validate register data:
    - Username: [A-Za-z0-9_]{4,16}
    - Email: [\w-\.]+@([\w-]+\.)+[\w-]{2,24}
    - Password: (?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,32} 
       Minimum eight characters, at least one letter and one number
- Also added hashing of passwords

<br><br>

✅ Small changes on database mappings to improve performance included:

- Users could only be identified by their email. This made it impossible to identify a user in the link for a profile page without leaking their email address.
- Created a UserID attribute which serves as the new primary key
- Additional adjustments on code wherever necessary afterwards

<br><br>

## Team Building

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/26575247-5891-49e8-9b38-b998c62326ca" height="250" />
<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/17359d97-4649-46a6-bc85-85da8f231cf4" height="250" />
<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/e402cd64-354d-4c57-8a15-91aebbee492b" height="250" />

<br><br>

See you next time!

**Your**   <img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1a218c37-d3b0-4c88-9ae1-827b5a4eba42">   **Team**

**Comment by ghost** [comment]
Great update, CookHub! The new frontend features, like the dialog popup with Carlos and the revamped settings and profile pages, sound fantastic and user-friendly. On the backend, the enhanced login and authentication logic, along with the database mappings, show your commitment to security and performance. Keep up the amazing work—looking forward to the result and demo!
